### ISSUE TITLE
Hashes of 4.39.4 have changed

### ISSUE THREAD

**Issue by gempir** [fonttools/fonttools#3120]
It looks like the hashes of the 4.39.4 was this a malicious act?

```
/root/python_requirements.txt (line 727)):\n        Expected sha256 106caf6167c4597556b31a8d9175a3fdc0356fdcd70ab19973c3b0d4c893c461\n        Expected     or dba8d7cdb8e2bac1b3da28c5ed5960de09e59a2fe7e63bb73f5a59e57b0430d2\n             Got        6581d950a47980cee84c3f453a366c31e30b5e508d6c9c7ece7b73924e63fe34\n\n\n
```

Looking at pypi all the .whl files have been uploaded recently (May 23) but the source files have stayed the same from the release of 4.39.4 on May 10

![image](https://github.com/fonttools/fonttools/assets/9765622/a09749b9-d316-4fd8-a16f-4fdd0ac20704)

**Comment by anthrotype** [comment]
yes, I did upload new wheel files yesterday (#3116) for the latest 4.39.4 release. They were built and uploaded by Github Actions from https://github.com/fonttools/fonttools-wheels, which checks out the fonttools repository as a submodule at the latest git tag. I did not make any changes to the source distribution so it is expected that for the current release 4.39.4 the source distribution is the same as before, only new binary wheels were uploaded.
Is that undesired for some reason? I don't see where the issue is coming from. Who/what tool alerted you that this might be "malicious"?

**Comment by gempir** [comment]
I generate hashes for python dependencies with like this:
```
pip-compile -v --generate-hashes --output-file=/tmp/python_requirements.txt --pip-args=--no-cache-dir /tmp/python_requirements.in
```
https://github.com/jazzband/pip-tools#using-hashes

That ensures I always grab the exact dependency of something. Every major package manager does the same yarn.lock, package-lock.json, composer.lock just to mention some examples. 

If you change the wheel files it would more sense to actually generate a new release IMO.

**Comment by anthrotype** [comment]
fair enough. I thought it'd be ok since there was no actual code changes, just a different way things gert packaged. Ok you may argue that the presence of native code (the cython extensions) where previously were only pure python modules does warrant a version bump change. I'll remove the wheel files from the current release and will make a new release soon with those.
Thanks for the heads up 👍

**Comment by gempir** [comment]
Hmm. I haven't published anything on pypi before, but is there maybe a pre-release of sorts that let's you not directly publish the newest version and only when all wheels are built and uploaded you can publish the release?

**Comment by anthrotype** [comment]
Maybe we don't need to have a separate repository just to have a distinct CI setup, I think I can add a new Github Actions yaml file in this same repository and set it up such that it does not run at every push but only upon git-tagged commits. That way I don't make the regular CI slower (it takes almost an hour to build all of them!). I think I'll do that before I make the next release.
